client/network/src/service.rs

```diff
@@ -1004,6 +988,40 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		Ok(())
 	}
 
+	/// Add peers to a peerset priority group.
+	///
+	/// Each `Multiaddr` must end with a `/p2p/` component containing the `PeerId`.
+	///
+	/// Returns an `Err` if one of the given addresses is invalid or contains an
+	/// invalid peer ID (which includes the local peer ID).
+	pub fn add_to_priority_group(&self, group_id: String, peers: HashSet<Multiaddr>) -> Result<(), String> {
```

Comment:
```suggestion
	pub async fn add_to_priority_group(&self, group_id: String, peers: HashSet<Multiaddr>) -> Result<(), String> {
```

Declaring this method as `async` would grant us a greater implementation flexibility in the future, e.g. a bounded channel between `NetworkService` and `NetworkWorker`.

@tomaka what do you think of requiring all new methods on `NetworkService` to be able to yield?

@ordian Are you able to `await` this method within Polkadot?

Reply:
yes, awaiting is fine, but would we need to change all three methods then (set_, add_, remove_)?

Reply:
Done in b1c28b2.

Reply:
:+1: As eventually it'd be nice if the queue from the service to the worker was bounded as well.